### PR TITLE
Fix include dir

### PR DESCRIPTION
**Pull request by holke**
**_Describe your changes here:_**

Continues previous PR #153.

This change applies to what happens when we `make install` t8code.
Previously all headers were put into `include/t8`.
Now, all headers are put into `include` directly respecting the subfolder structure of t8code.
Thus, if a header is in the source folder `src/folder/header.h` then it is installed to `include/folder/header.h`.

The folder structure of `include/` should now mirror the folder structure of `src/` (ignoring sc and p4est files).

**_All these boxes must be checked by the reviewers before merging the pull request:_**

- [x] The author added a BSD statement to `doc/` (or already has one)
- [x] The code compiles without warning in debugging and release mode, with and without MPI (this should be executed automatically in a github action)

  If the Pull request indroduces code that is not covered by the github action (for example coupling with a new library):
  - [ ] Should this use case be added to the github action?
  - [ ] If not, does the specific use case compile and all tests pass (check manually)

- [x] All tests pass (in various configurations, this should be executed automatically in a github action)
- [ ] New source/header files are properly added to the Makefiles
- [x] The reviewer executed the new code features at least once and checked the results manually
- [ ] The code is covered in an existing or new test case
- [ ] The code follows the [t8code coding guidelines](https://github.com/holke/t8code/wiki/Coding-Guideline)
- [ ] The code is well documented
- [ ] All function declarations, structs/classes and their members have a proper doxygen documentation
- [ ] All new algorithms and data structures are sufficiently optimal in terms of memory and runtime (If this should be merged, but there is still potential for optimization, create a new issue)
- [ ] Testing of this template: If you feel something is missing from this list, contact the developers

### CODE REVIEW

File: src/Makefile.am [hunk 1]
```diff
@@ -2,31 +2,62 @@
...
+libt8_installed_headers_geometry_impl = \
   src/t8_geometry/t8_geometry_implementations/t8_geometry_linear.h \
-  src/t8_geometry/t8_geometry_helpers.h \
   src/t8_geometry/t8_geometry_implementations/t8_geometry_analytic.hxx \
   src/t8_geometry/t8_geometry_implementations/t8_geometry_linear.hxx \
-  src/t8_geometry/t8_geometry_implementations/t8_geometry_zero.hxx \
-  src/t8_vec.h src/t8_vtk.h \
-  src/t8_forest/t8_forest_iterate.h src/t8_forest/t8_forest_partition.h
+  src/t8_geometry/t8_geometry_implementations/t8_geometry_zero.hxx
+libt8_installed_headers_schemes_default =
```
Comment (Davknapp) [inline src/Makefile.am:51]: Why are these empty?

**Review by Davknapp — COMMENTED** [review]
Please address my question, I do not understand that part of the Makefile.
Apart from that it is ready to merge!

File: src/Makefile.am (hunk 1 above)
Comment (Davknapp) [reply src/Makefile.am:51]: Thx for the explanation! At first, I thaught it was a doubling.

**Review by Davknapp — CHANGES_REQUESTED** [review]
Please solve the remaining merge-conflicts, then we are ready to merge!

**Comment by holke** [comment]
Done

**Review by Davknapp — APPROVED** [review]

**Pull request merged**